main/src/cgeo/geocaching/settings/Settings.java

```diff
@@ -1962,22 +2004,47 @@ public static int getListInitialLoadLimit() {
         return sensitiveKeys;
     }
 
-    public static void putLogTemplate(final String preferenceBaseName, final String title, final String text) {
-        putStringDirect(preferenceBaseName + ".Title", title);
-        putStringDirect(preferenceBaseName, text);
+    public static void putLogTemplate(final PrefLogTemplate template) {
+        List<PrefLogTemplate> templates = getLogTemplates();
+        int templateIndex = templates.indexOf(template);
+        if (templateIndex == -1 && template.getText() == null) {
+            return;
+        } else if (templateIndex > -1 && template.getText() == null) {
+            templates.remove(templateIndex);
+        } else if (templateIndex == -1) {
+            templates.add(template);
+        } else {
+            templates.set(templateIndex, template);
+        }
+        try {
+            putString(R.string.pref_logTemplates, MAPPER.writeValueAsString(templates));
+        } catch (JsonProcessingException e) {
+            Log.e("Failure writing log templates: "+e.getMessage());
+        }
     }
 
-    public static ImmutablePair<String, String> getLogTemplate(final String preferenceBaseName) {
-        return new ImmutablePair<>(getStringDirect(preferenceBaseName + ".Title", ""), getStringDirect(preferenceBaseName, ""));
+    @Nullable
+    public static PrefLogTemplate getLogTemplate(final String key) {
+        for (PrefLogTemplate template : getLogTemplates()) {
+            if (template.getKey().equals(key)) {
+                return template;
+            }
+        }
+        return null;
     }
 
-    public static String getLogTemplatePreview(final String preferenceBaseName) {
-        final ImmutablePair<String, String> template = getLogTemplate(preferenceBaseName);
-        return template.getKey() + ": " + template.getValue();
+    @Nullable
+    public static List<PrefLogTemplate> getLogTemplates() {
+        List<PrefLogTemplate> templates = null;
```

Comment:
Probably we should rather return an empty list instead of `null`, as other code places doesn't seem to check the null case correctly.